src/pytorch_lightning/lite/wrappers.py

```diff
@@ -115,6 +118,26 @@ def _convert_float_tensor(t: Tensor) -> Tensor:
         output = apply_to_collection(output, function=_convert_float_tensor, dtype=Tensor)
         return output
 
+    @overload
+    def state_dict(self, *, destination: T_destination, prefix: str = ..., keep_vars: bool = ...) -> T_destination:
+        ...
+
+    @overload
+    def state_dict(self, *, prefix: str = ..., keep_vars: bool = ...) -> T_destination:
+        ...
```

Comment:
Why do we define these overloads? I tried removing them and doesn't seem like anything changes

Reply:
this does help a lot with IDE completion and should enable us to remove the typing of the method's implementation at all as this already defines typing.

Reply:
I'm sorry, I can't remove the types @justusschock, our mypy says: Missing type annotation for function etc.

> Why do we define these overloads? I tried removing them and doesn't seem like anything changes

@carmocca I get this: 

```
src/pytorch_lightning/lite/wrappers.py:121: error: Signature of "state_dict" incompatible with supertype "Module"  [override]
src/pytorch_lightning/lite/wrappers.py:121: note:      Superclass:
src/pytorch_lightning/lite/wrappers.py:121: note:          @overload
src/pytorch_lightning/lite/wrappers.py:121: note:          def [T_destination <: Dict[str, Any]] state_dict(self, *, destination: T_destination, prefix: str = ..., keep_vars: bool = ...) -> T_destination
src/pytorch_lightning/lite/wrappers.py:121: note:          @overload
src/pytorch_lightning/lite/wrappers.py:121: note:          def state_dict(self, *, prefix: str = ..., keep_vars: bool = ...) -> Dict[str, Any]
src/pytorch_lightning/lite/wrappers.py:121: note:      Subclass:
src/pytorch_lightning/lite/wrappers.py:121: note:          def [T_destination <: Dict[str, Any]] state_dict(self, destination: Optional[T_destination] = ..., prefix: str = ..., keep_vars: bool = ...) -> Optional[T_destination]

```

@carmocca @justusschock feel free to push any changes regarding typing, as long as it is valid. I don't care too much about the typing details here.